conda/base/context.py

```diff
@@ -955,7 +955,12 @@ def binstar_upload(self):
 
     @property
     def verbosity(self):
-        return 2 if self.debug else self._verbosity
+        #                   0 = logging.WARN, standard output
+        #           -v    = 1 = logging.WARN, detailed output
+        #           -vv   = 2 = logging.INFO
+        # --debug = -vvv  = 3 = logging.DEBUG
+        #           -vvvv = 4 = logging.TRACE
+        return 3 if self.debug else self._verbosity
```

Comment:
This doesn't actually conform with the logging module, `logging.WARN` was long renamed to `logging.WARNING` and the levels are two digit integers.

```suggestion
        #                    0 = logging.WARNING, standard output
        #           -v    = 10 = logging.WARNING, detailed output
        #           -vv   = 20 = logging.INFO
        # --debug = -vvv  = 30 = logging.DEBUG
        #           -vvvv = 40 = logging.TRACE
        return logging.DEBUG if self.debug else self._verbosity
```

BTW `logging.TRACE` is only added in `conda/gateways/logging.py`, so technically isn't `4` or `40` but `5`!

Reply:
A `--trace` CLI options (even if hidden), might be useful as well.

Reply:
The verbosity values are simply a count of the number of `-v`s present (see our custom `conda.cli.conda_argparse.NullCountAction`). So my comments here are better written as:

```suggestion
        #                   0 → logging.WARN, standard output
        #           -v    → 1 → logging.WARN, detailed output
        #           -vv   → 2 → logging.INFO
        # --debug = -vvv  → 3 → logging.DEBUG
        #           -vvvv → 4 → logging.TRACE
        return 3 if self.debug else self._verbosity
```

What if we leave `context.verbosity` as is (make it easy to say `if context.verbose`) and then introduce a new `context.log_level` that maps the verbosity to the logging level?

Reply:
> What if we leave context.verbosity as is (make it easy to say if context.verbose) and then introduce a new context.log_level that maps the verbosity to the logging level?

I like that!